src/pages/cli/project/troubleshooting.mdx

```diff
@@ -0,0 +1,180 @@
+1. amplify pull --appId successfulEnvAppID --envName successfulEnv
+2. amplify env checkout failingEnv
+3. amplify push 
+
+If the push fails, please check the CloudFormation event logs for the failing resource to get detailed information and use this to fix issues in the environment. Visit the CloudFormation service console and search for you application name. e.g. An application named lil deployed in the beta environment would have a CloudFormation stack named amplify-lil-beta-{random chars}. Click on stack name and the Resources tab to view the status and the error reason.
+This figure displays the CloudFormation status logs for a failed resource deployed through Amplify CLI:
+###### fig.1.1 Cloudformation Stack Deployment Error
+![ts-guide-cfn-error](/images/ts-guide-cfn-error.png)
+
+### Handling push failures from: Application resources hitting AWS Service Limits
+<Callout information>
+Error Messages: 
+"Limit on the number of resources in a single stack operation exceeded” 
+"Cannot exceed quota for PoliciesPerRole"
+</Callout>
+
+AWS maintains service quotas (formerly called service limits) for each account to help guarantee the availability of AWS resources and prevent accidental provisioning of more resources than needed. If your application contains many resources, especially for particularly big GraphQL schemas with complex connections and resolvers or REST APIs with many resolvers, it’s likely that you might eventually hit AWS Service Limits leading to amplify push failures. 
+To identify if a push failure is because of exceeding [AWS Service level quotas](https://console.aws.amazon.com/servicequotas) please check the AWS console for the CloudFormation deployment/failure logs.
+
+##### Suggested Resolution: 
+1. If the bug is related to hitting service-quotas, you can use AWS Service Quotas console to view and request increases for most AWS quotas.
+2. If your application uses Amplify DataStore, some resource limits may affect deployment. Referring the [Amplify DataStore Best Practices](https://docs.aws.amazon.com/whitepapers/latest/amplify-datastore-implementation/amplify-datastore-best-practices.html) document may provide insights into making the correct design decisions to help resolve these types of issues. 
+3. If the service quotas cannot be increased, then this may require you to redesign some aspects of your application for sustainable running in a serverless environment. Referring to [AWS Knowledge Center](https://aws.amazon.com/premiumsupport/knowledge-center/) may help understand the different prescribed design changes, for example, A complex GraphQL schema could be simplified to reduce the number of GraphQL resources in use e.g Collapsing Models, Composite Keys, Lambda resolvers. Using escape hatches like overrides and export may help to scale the application beyond the CLI generated architecture. 
```

Comment:
```suggestion
3. If the service quotas cannot be increased, then this may require you to redesign some aspects of your application for sustainable running in a serverless environment. Referring to [AWS Knowledge Center](https://aws.amazon.com/premiumsupport/knowledge-center/) may help understand the different prescribed design changes. For example, a complex GraphQL schema could be simplified to reduce the number of GraphQL resources in use e.g collapsing models, composite keys, Lambda resolvers. Using escape hatches like overrides and export may help to scale the application beyond the CLI generated architecture. 
```